### DISCUSSION TITLE
STM32G030F6Pxx - tone library/ Tone Melody example not working

### DISCUSSION THREAD

**Discussion by vik0asth** [stm32duino/Arduino_Core_STM32#1627] [Libraries]
Hi,
I have a custom board with STM32G0F6Px MCU. I am trying to run tone example on that but not getting any output/ sound on buzzer.
I saw the support for this MCU is added in latest version now and the clock file & ldscript is same as what I generated for the board using STM32 IDE.

Same tone example code runs fine on Nucleo-64 STM32G071RB eval board but not on custom board. The "variant_generic.h" files for the two MCUs uses two different timers - could that be the problem? TIM6/TIM7 are not supported on STM32G0F6Px, so which timer to use?

![image](https://user-images.githubusercontent.com/2757091/149842461-f4281938-ef6c-4cb7-a04e-a8a07b7c433e.png)

Can anyone please help to resolve this issue?

Thanks in advance.

**Comment by ABOSTM** [comment]
Hi @vik0asth,
Note that Nucleo-64 STM32G071RB uses variant_NUCLEO_G071RB.h and not variant_generic.h.
Whatever, that is right that Nucleo STM32G071RB use TIM6 for Tone.
And as you noticed TIM6 doesn't exist on STM32G030F6Px, so it uses TIM14 in  "variant_generic.h". That is fine to use TIM14 no need to change.

I don't have STM32G030, but I tested my Nucleo STM32G071RB with TIM14 and tone example works.
Could you check that the pin used in the Tone sketch match your pin used for tone on your custom board ?
